include/helpers/debugger.hpp

```diff
@@ -17,7 +17,14 @@ class Debugger {
   void pokeMemory(size_t varSize, u64 address, u64 value);
   MemoryInfo queryMemory(u64 address);
   void readMemory(void *buffer, size_t bufferSize, u64 address);
-  void writeMemory(void *buffer, size_t bufferSize, u64 address);
+  bool writeMemory(void *buffer, size_t bufferSize, u64 address);
+
+  template<typename T>
+  bool WriteMemory(u64 address, const T& value)
```

Comment:
The issue with this function is, what if the user wants to edit a u32 value stored in memory to for example 0x12. Before the edit, the value in memory was maybe 0xFFFFFFFF. So when calling this function with 0x12, the new value in memory now is 0x12FFFFFF which isn't exactly what the user would expect

Reply:
0x12 is still a 32bit value. `WriteMemory(..., (uint8_t)0x12)` would produce your result.

Reply:
I don't know how familiar you are with templates, but T has to be a known type at the compile time. It can't be used for any types later, but for this one you already defined.
So if you give it a u32, it will write only u32.
Look how I've used it to fix the float problem.  

It's one line of code:
`m_debugger->WriteMemory(address, static_cast<float>(std::atof(input)));`
istead of
```cpp
auto value = static_cast<float>(std::atof(input));
m_debugger->writeMemory(&value, dataTypeSizes[m_searchType], address);
```
Also since the compiler knows that he will get a float there, he already knows the size without getting it later from dataTypeSizes[m_searchType], it's also better optimized.

Reply:
You're right. That was a stupid example.
But since the value to write will always be of the same type in the end, a int, it's gonna write a 4 byte value like this even if you only want to write one or two bytes. You need to give it the size the user selected beforehand

Reply:
This for you still have the 
`Result writeMemory(void *buffer, size_t bufferSize, u64 address);`
function. 🤔

Reply:
I'm just confused why to use a template for that and then only use it for floats and doubles. I think it would be better to use the same function as before but have a check to reinterpret_cast the floats into ints beforehand instead of static_casting them implicitly like before

Reply:
Well, I wanted to improve the code bit by bit and chance the architecture in some places to make it easier.
I've removed it now.